src/operators/ExtrudeStacked/avtExtrudeStackedFilter.C

```diff
@@ -0,0 +1,2272 @@
+//  Method:      avtExtrudeStackedFilter::SetAtts
+//
+//  Purpose:
+//      Sets the state of the filter based on the attribute object.
+//
+//  Arguments:
+//      a        The attributes to use.
+//
+//  Programmer: Allen Sanderson
+//  Creation:   August 31, 2023
+//
+// ****************************************************************************
+
+void
+avtExtrudeStackedFilter::SetAtts(const AttributeGroup *a)
+{
+    atts = *(const ExtrudeStackedAttributes*) a;
+
+    avtVector axis(atts.GetAxis());
+
+    if (axis.x == 0. && axis.y == 0. && axis.z == 0.)
+    {
+        EXCEPTION1(BadVectorException, "ExtrudeStacked - Zero length extrusion axis.");
+    }
+
+    if(atts.GetLength() == 0.)
+    {
+      if( atts.GetByVariable() ) {
+        EXCEPTION1(ImproperUseException, "ExtrudeStacked - The scale cannot be 0.");
+      }
+      else {
+        EXCEPTION1(ImproperUseException, "ExtrudeStacked - The length cannot be 0.");
+      }
+    }
+
+    if(atts.GetSteps() < 1)
+    {
+        EXCEPTION1(ImproperUseException, "ExtrudeStacked - The number of steps must be at least 1.");
+    }
+}
+
+
+// ****************************************************************************
+//  Method: avtExtrudeStackedFilter::ModifyContract
+//
+//  Purpose:
+//      This method makes any necessay modification to the VisIt contract
+//      to request that the extrude variable is also loaded if it
+//      is not already part of the contract.
+//
+//  Arguments:
+//      in_contract   The current contract
+//
+//  Programmer: Allen Sanderson
+//  Creation:   August 31, 2023
+//
+// ****************************************************************************
+
+avtContract_p
+avtExtrudeStackedFilter::ModifyContract(avtContract_p in_contract)
+{
+    avtDataRequest_p in_dr = in_contract->GetDataRequest();
+    std::string var = in_dr->GetOriginalVariable();
+
+    // Create the output contract.
+    avtContract_p out_contract = nullptr;
+
+    // If extruding by variable(s) add the variable(s) to the contract
+    // if necessay.
+    if( atts.GetByVariable() && atts.GetScalarVariableNames().size() )
+    {
+      if (!atts.AttributesAreConsistent())
+      {
+        debug3 << "ExtrudeStacked operator attributes are inconsistent." << endl;
+        return in_contract;
+      }
+
+      const char *inPipelineVar = in_contract->GetDataRequest()->GetVariable();
+      std::string outPipelineVar(inPipelineVar);
+
+      stringVector curVariableVarNames = atts.GetScalarVariableNames();
+      stringVector needSecondaryVars;
+
+      for (size_t i=0; i<curVariableVarNames.size(); ++i)
+      {
+          if (curVariableVarNames[i] == "default")
+          {
+            curVariableVarNames[i] = inPipelineVar;
+            break;
+          }
+      }
+
+      // If multiple extrustions see if the current output variable is
+      // in the list of variables.
+      int variableCount = curVariableVarNames.size();
+#if defined(REPLICATE_EXTRUDE_OPERATOR)
```

Comment:
Why is this here? When or why would it be defined?

Reply:
The define is a future addition. The stacked version can do everything the current extrude operator and more. However, with a single variable the display by (color) behavior is slightly different. So it that flag allows one to replicate the extrude operator exactly.

Reply:
I have flipped flopped on this flag. By default it is now enabled in the Qvis.  The reason being is that I replicated the ability to extrude by one variable and color by the original data in a manner that makes sense when there is a single variable. It does not make sense when there are multiple variable extrusions.  The "Display by" option is updated based on one or more variables to select "Original data" or "Index." That should make it clear to the user what is happening.

Also I discovered the current fixed length extrude is limited to the z axis. That may or may not make sense. It also was limited to 2D XY plane data, cold not do a curve, or a point. I expanded all options in all directions.  As such, the the fixed option is in place so a user can have additional options.